### PR TITLE
Revisit equivariance

### PR DESCRIPTION
**Pull request by rousseab**
In this PR I centralize the tests pertaining to equivariance, which are now performed exclusively on `ScoreNetwork` objects.

In summary:
- I fix "bugs" in the `MACE`-related architectures where we were previously a bit fast and dirty when distinguishing `cartesian_scores` and `coordinates_scores`. This only matters for non-cubic cells, which we have not been using so far. I *claim* that this "bug" has had no serious impact on our previous studies.
- The `EGNN` structures no longer 'pass too many tests' as I mentioned verbally in the past. That being said, it should be remembered that our current implementation of `EGNN` depends on the assumption that we have a cubic unit cell: the architecture (and the formalism!) must be reviewed for a non-cubic unit cell. 

I also remove a few (now useless) tests that were checking equivariance at a level other than the `ScoreNetwork`.

### CODE REVIEW

File: src/diffusion_for_multi_scale_molecular_dynamics/models/score_networks/score_network.py
Comment (sblackburn86) [inline src/diffusion_for_multi_scale_molecular_dynamics/models/score_networks/score_network.py:8]: if you use an "r" string, you should use the equation markup
... math::
Otherwise, remove the "r" string

File: tests/models/score_network/test_score_network_equivariance.py
Comment (sblackburn86) [inline tests/models/score_network/test_score_network_equivariance.py:380]: typo: rational -> rotational

File: tests/models/score_network/test_score_network_equivariance.py
Comment (sblackburn86) [inline tests/models/score_network/test_score_network_equivariance.py:391]: typo

File: tests/models/score_network/test_score_network_equivariance.py
Comment (sblackburn86) [inline tests/models/score_network/test_score_network_equivariance.py:429]: I am not sure, I would need to think more about it. It's fine to leave as a TODO for now

**Review by sblackburn86 — APPROVED** [review]

File: src/diffusion_for_multi_scale_molecular_dynamics/models/score_networks/score_network.py
Comment (rousseab) [reply src/diffusion_for_multi_scale_molecular_dynamics/models/score_networks/score_network.py:8]: done.

File: tests/models/score_network/test_score_network_equivariance.py
Comment (rousseab) [reply tests/models/score_network/test_score_network_equivariance.py:380]: good catch :D

File: tests/models/score_network/test_score_network_equivariance.py
Comment (rousseab) [reply tests/models/score_network/test_score_network_equivariance.py:391]: fixed.

File: tests/models/score_network/test_score_network_equivariance.py
Comment (rousseab) [reply tests/models/score_network/test_score_network_equivariance.py:429]: Sounds good.

**Comment by rousseab** [comment]
I'll wait until you merge your PR before trying to merge this. I'll manage the conflicts, if any.

**Pull request merged**
